.gen/proto/api/v1/domain.pb.go

```diff
@@ -1,112 +1,159 @@
-// copies or substantial portions of the Software.
-// 
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
 
-// Code generated by protoc-gen-gogo. DO NOT EDIT.
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// versions:
+// 	protoc-gen-go v1.25.0
+// 	protoc        v3.13.0
 // source: uber/cadence/api/v1/domain.proto
 
-package apiv1
+package v1
 
 import (
-	fmt "fmt"
-	io "io"
-	math "math"
-	math_bits "math/bits"
 	reflect "reflect"
-	strconv "strconv"
-	strings "strings"
-	time "time"
-
-	_ "github.com/gogo/protobuf/gogoproto"
-	proto "github.com/gogo/protobuf/proto"
-	github_com_gogo_protobuf_sortkeys "github.com/gogo/protobuf/sortkeys"
-	github_com_gogo_protobuf_types "github.com/gogo/protobuf/types"
-	_ "github.com/golang/protobuf/ptypes/duration"
-	_ "github.com/golang/protobuf/ptypes/timestamp"
+	sync "sync"
+
+	proto "github.com/golang/protobuf/proto"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
 )
 
-// Reference imports to suppress errors if they are not otherwise used.
-var _ = proto.Marshal
-var _ = fmt.Errorf
-var _ = math.Inf
-var _ = time.Kitchen
+const (
+	// Verify that this generated code is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(20 - protoimpl.MinVersion)
+	// Verify that runtime/protoimpl is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
+)
 
-// This is a compile-time assertion to ensure that this generated file
-// is compatible with the proto package it is being compiled against.
-// A compilation error at this line likely means your copy of the
-// proto package needs to be updated.
-const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
+// This is a compile-time assertion that a sufficiently up-to-date version
+// of the legacy proto package is being used.
+const _ = proto.ProtoPackageIsVersion4
 
 type DomainStatus int32
 
 const (
-	DOMAIN_STATUS_INVALID    DomainStatus = 0
-	DOMAIN_STATUS_REGISTERED DomainStatus = 1
-	DOMAIN_STATUS_DEPRECATED DomainStatus = 2
-	DOMAIN_STATUS_DELETED    DomainStatus = 3
+	DomainStatus_DOMAIN_STATUS_INVALID    DomainStatus = 0
```

Comment:
One annoying thing with the official compiler, is that it prefix enum values with enum type. This results in duplicative prefix, as we already do that in .proto files to comply with `buf lint` rules. We could either:
- Ignore that (not that big a deal, since they will be mapped to internal types)
- Drop prefix in proto files. Not ideal, as this would require to deviate from default lint rules. Also _INVALID value would still need a prefix, as they the same enum value can not appear within single proto package.
- Use custom post-generation script to cut unwanted prefix away.

Reply:
hmmmmm.  tbh I wonder if this is code-generator-specific: https://docs.buf.build/lint-rules#enum_value_prefix
>Protobuf enums use C++ scoping rules, which makes it not possible to have two enums in the same package with the same enum value name (an exception is when enums are nested, in which case this rule applies within the given message).

That's accurate for the v1 generator, but not v2 (since it adds this prefix).  And it could go either way for other languages.

On the theory that other languages may use the public API proto files, I think I'll vote to keep it this verbose nonsense.  Once mapped to internal types it really doesn't matter.

Reply:
Yes this is indeed code generator specific. Gogo compiler does not do this by default and also have option to change the behavior. Official protoc-gen-go however does not support it. It is clearly visible from [source](https://github.com/protocolbuffers/protobuf-go/blob/master/compiler/protogen/protogen.go#L645). There is an open issue, but looks like that it is unlikely to be resolved: https://github.com/golang/protobuf/issues/513

Linter is doing good job here, as it really depends on the compiler.

Reply:
Yes this is indeed code generator specific. Gogo compiler does not do this by default and also have option to change the behavior. Official protoc-gen-go however does not support it. It is clearly visible from [source](https://github.com/protocolbuffers/protobuf-go/blob/master/compiler/protogen/protogen.go#L645). There is an open issue: https://github.com/golang/protobuf/issues/513

Linter is doing good job here, as it really depends on the compiler.